src/store/modules/ProfilesModule.ts

```diff
@@ -0,0 +1,51 @@
+import FileUtils from "../../utils/FileUtils";
+import R2Error from "../../model/errors/R2Error";
+import { ActionTree, Store } from "vuex";
+import { State as RootState } from "../../store";
+import Profile from "../../model/Profile";
+import FsProvider from "../../providers/generic/file/FsProvider";
+
+interface State {
+    profileList: string[];
+}
+
+/**
+ * State for Profiles, i.e. list for profiles in a single game/community.
+ */
+export const ProfilesModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        profileList: ['Default'],
+    }),
+    mutations: {
+        setProfileList(state: State, profileList: string[]) {
+            state.profileList = profileList;
+        },
+    },
+    actions: <ActionTree<State, RootState>>{
+        async removeSelectedProfile({commit, rootGetters, state, dispatch}, params: {fs: FsProvider, store: Store<any>}) {
+            const activeProfile: Profile = rootGetters['profile/activeProfile'];
+            const path = activeProfile.getPathOfProfile();
+            const profileName = activeProfile.getProfileName();
+
+            try {
+                await FileUtils.emptyDirectory(path);
+                await params.fs.rmdir(path);
+            } catch (e) {
+                const err = R2Error.fromThrownValue(e, 'Error whilst deleting profile from disk');
+                await params.store.dispatch('error/handleError', err);
+            }
+
+            if (profileName.toLowerCase() !== 'default') {
+                try {
+                    state.profileList = state.profileList.filter((p: string) => p !== profileName)
+                    await params.store.dispatch('profile/updateActiveProfile', 'Default');
```

Comment:
In case the store parameter was passed to make this line work, this approach could be used: https://vuex.vuejs.org/guide/modules.html#accessing-global-assets-in-namespaced-modules